pkg/scripts/render.go

```diff
@@ -210,39 +207,49 @@ var (
 			sudo yum install -y containerd.io-%s
 			sudo yum versionlock add containerd.io
 
-			{{ template "containerd-config" . -}}
+			{{ template "containerd-systemd-setup" . -}}
 			`,
 			defaultContainerdVersion,
 		),
 
 		"yum-containerd-amzn": heredoc.Docf(`
+			{{ template "container-runtime-daemon-config" . }}
 			sudo yum versionlock delete containerd cri-tools || true
 			sudo yum install -y containerd-%s cri-tools-%s
 			sudo yum versionlock add containerd cri-tools
 
-			{{ template "containerd-config" . -}}
+			{{ template "containerd-systemd-setup" . -}}
 			`,
 			defaultAmazonContainerdVersion,
 			defaultAmazonCrictlVersion,
 		),
 
 		"flatcar-containerd": heredoc.Doc(`
-			cat <<EOF | sudo tee /etc/crictl.yaml
-			runtime-endpoint: unix:///run/containerd/containerd.sock
-			EOF
+			{{ template "container-runtime-daemon-config" . }}
+			{{ template "containerd-systemd-environment" . }}
 
-			sudo mkdir -p /etc/systemd/system/containerd.service.d
-			cat <<EOF | sudo tee /etc/systemd/system/containerd.service.d/environment.conf
+			cat <<EOF | sudo tee /etc/systemd/system/containerd.service.d/10-kubeone.conf
 			[Service]
 			Restart=always
-			EnvironmentFile=-/etc/environment
```

Comment:
Why have you removed this? Isn't this used in the proxy setups?

Reply:
it's extracted in individual (but shared) template where environment.conf is set